### ISSUE TITLE
Speedup opening tables

### ISSUE THREAD

**Issue by killme2008** [GreptimeTeam/greptimedb#353]
When there were many tables( over 200 for example), the datanode startup too slowly(nearly half of a minute).

I think we should turn this performance by opening table in parallel or optimize the replay processing etc.

**Comment by v0y4g3r** [comment]
As we've already introduced raft-engine based WAL in #799, we may have a significant improvement in startup speed, because replaying entries in some namespace no longer needs to read the whole WAL. 

I did a simple validation by creating 1000 tables:
```bash
while [  $COUNTER -lt 1000 ]; do
    mysql -h 127.0.0.1 -P 4002 -e "CREATE TABLE monitor${COUNTER} (host STRING,ts TIMESTAMP,cpu DOUBLE DEFAULT 0,memory DOUBLE,TIME INDEX (ts),PRIMARY KEY(host)) ENGINE=mito WITH(regions=1);"
    let COUNTER=COUNTER+1
done
```

And then restart datanode:
```
2023-01-06T06:26:34.657599Z  INFO cmd::standalone: Standalone frontend options: FrontendOptions ....
....
2023-01-06T06:26:35.892063Z  INFO catalog::local::manager: All system catalog entries processed, max table id: 2023
```

It takes about 1235ms to recovering all 1000 tables. The performance looks fairly acceptable, but I think we can further speed up table recovering by parallel opening table.

**Comment by killme2008** [comment]
@v0y4g3r  Awesome!

**Comment by killme2008** [comment]
Implemented in #1392 #1290
